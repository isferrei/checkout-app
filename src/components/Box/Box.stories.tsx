import type { Meta, StoryObj } from "@storybook/react";

import { Box } from "./Box";

const meta = {
  title: "Example/Box",
  component: Box,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
} satisfies Meta<typeof Box>;

export default meta;
type Story = StoryObj<typeof Box>;

export const Default = {
  args: {
    variation: "shadow",
    children: "Lorem Ipsum",
  },
} satisfies Story;
