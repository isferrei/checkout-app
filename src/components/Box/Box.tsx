/* eslint-disable react/Box-has-type */
import classNames from "classnames";
import type { ReactNode } from "react";

type BoxTheme = "shadow" | "default";

type BoxProps = {
  variation: BoxTheme;
  height?: number;
  width?: number;
  bg?: string;
  children: ReactNode;
};

const Box = (props: BoxProps) => {
  const handleThemes = (BoxTheme: BoxTheme) => {
    switch (BoxTheme) {
      case "shadow":
        return "drop-shadow-[0_4px_20px_rgba(0,0,0,0.05)]";
      default:
        return "";
    }
  };

  return (
    <div
      {...props}
      style={{
        height: props.height ? `${props.height}rem` : "5rem",
        width: props.width ? `${props.width}rem` : "23.75rem",
        background: props.bg ? props.bg : "#fff",
      }}
      className={classNames(
        "flex justify-center items-center w-full rounded-[15px] text-[0.94rem]",
        handleThemes(props.variation)
      )}
    >
      {props.children}
    </div>
  );
};

export { Box };
