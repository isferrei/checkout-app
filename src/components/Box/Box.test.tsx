import { render } from "@testing-library/react";
import { Box, BoxProps } from "./Box";

describe("Box component", () => {
  const renderBox = (props: BoxProps) => {
    return render(<Box {...props} />);
  };

  test("renders with default props", () => {
    const { container } = renderBox({
      variation: "default",
      children: "Content",
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders with shadow variation", () => {
    const { container } = renderBox({
      variation: "shadow",
      children: "Content",
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders with custom styles", () => {
    const { container } = renderBox({
      variation: "default",
      children: "Content",
      height: 10,
      width: 20,
      bg: "#ccc",
      className: "custom-class",
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
