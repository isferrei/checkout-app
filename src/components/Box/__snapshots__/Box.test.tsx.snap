// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component renders with custom styles 1`] = `
<div
  bg="#ccc"
  class="flex justify-center items-center w-full rounded-[15px] text-[0.94rem] custom-class"
  data-testid="box"
  height="10"
  style="height: 10rem; width: 20rem; background: rgb(204, 204, 204);"
  variation="default"
  width="20"
>
  Content
</div>
`;

exports[`Box component renders with default props 1`] = `
<div
  class="flex justify-center items-center w-full rounded-[15px] text-[0.94rem] undefined"
  data-testid="box"
  style="background: rgb(255, 255, 255);"
  variation="default"
>
  Content
</div>
`;

exports[`Box component renders with shadow variation 1`] = `
<div
  class="flex justify-center items-center w-full rounded-[15px] text-[0.94rem] undefined drop-shadow-[0_4px_20px_rgba(0,0,0,0.05)]"
  data-testid="box"
  style="background: rgb(255, 255, 255);"
  variation="shadow"
>
  Content
</div>
`;
